From 253692b6f389825473fdbbbc93c587901c3722cb Mon Sep 17 00:00:00 2001
From: Josef Schlehofer <pepe.schlehofer@gmail.com>
Date: Mon, 25 Nov 2019 13:51:38 +0100
Subject: [PATCH] Fix build for OpenWrt

Fix:
In file included from src/tvheadend.h:31:0,
                  from src/intlconv.c:2:
/build/staging_dir/toolchain-arm_cortex-a9+vfpv3_gcc-7.3.0_musl_eabi/include/fortify/string.h: In function 'mempcpy':
/build/staging_dir/toolchain-arm_cortex-a9+vfpv3_gcc-7.3.0_musl_eabi/include/fortify/string.h:149:9: error: called object '__orig_mempcpy' is not a function or function pointer
return __orig_mempcpy(__d, __s, __n);

---
 src/tvheadend.h | 15 +++++----------
 1 file changed, 5 insertions(+), 10 deletions(-)

diff --git a/src/tvheadend.h b/src/tvheadend.h
index e9ec6bbe3..04642263d 100644
--- a/src/tvheadend.h
+++ b/src/tvheadend.h
@@ -20,8 +20,6 @@

 #include "build.h"

-#define _GNU_SOURCE
-#include <pthread.h>
 #include <stdlib.h>
 #include <stdio.h>
 #include <errno.h>
@@ -32,9 +30,6 @@
 #include <assert.h>
 #include <unistd.h>
 #include <limits.h>
-#if ENABLE_LOCKOWNER || ENABLE_ANDROID
-#include <sys/syscall.h>
-#endif
 #include "queue.h"
 #include "tvh_string.h"
 #include "hts_strtab.h"
@@ -410,7 +405,7 @@ typedef enum {
   /**
    * Packet with data.
    *
-   * sm_data points to a th_pkt. th_pkt will be unref'ed when
+   * sm_data points to a th_pkt. th_pkt will be unref'ed when
    * the message is destroyed
    */
   SMT_PACKET,
@@ -458,7 +453,7 @@ typedef enum {
    *
    * End of streaming. If sm_code is 0 this was a result to an
    * unsubscription. Otherwise the reason was external and the
-   * subscription scheduler will attempt to start a new streaming
+   * subscription scheduler will attempt to start a new streaming
    * session.
    */
   SMT_STOP,
@@ -605,7 +600,7 @@ typedef struct streaming_target {
  *
  */
 typedef struct streaming_queue {
-
+
   streaming_target_t sq_st;

   pthread_mutex_t sq_mutex;    /* Protects sp_queue */
@@ -613,7 +608,7 @@ typedef struct streaming_queue {

   size_t          sq_maxsize;  /* Max queue size (bytes) */
   size_t          sq_size;     /* Actual queue size (bytes) - only data */
-
+
   struct streaming_message_queue sq_queue;

 } streaming_queue_t;
@@ -804,7 +799,7 @@ void sha1_calc(uint8_t *dst, const uint8_t *d1, size_t d1_len, const uint8_t *d2
 uint32_t gcdU32(uint32_t a, uint32_t b);
 static inline int32_t deltaI32(int32_t a, int32_t b) { return (a > b) ? (a - b) : (b - a); }
 static inline uint32_t deltaU32(uint32_t a, uint32_t b) { return (a > b) ? (a - b) : (b - a); }
-
+
 #define SKEL_DECLARE(name, type) type *name;
 #define SKEL_ALLOC(name) do { if (!name) name = calloc(1, sizeof(*name)); } while (0)
 #define SKEL_USED(name) do { name = NULL; } while (0)
--
2.24.0
